app/controllers/bookmarks_controller.rb

```diff
@@ -102,15 +102,24 @@ def index
           # Remove conditional and call to BookmarkSearch
           if use_new_search?
             @search = BookmarkSearchForm.new(options.merge(faceted: true, parent: @owner))
-            # For listing all of the tags that it's possible to filter on within
-            # an owner's bookmarks page
-            @filtering_facets = BookmarkSearchForm.new(base_options.merge({parent: @owner})).search_results.facets
           else
             @search = BookmarkSearch.new(options.merge(faceted: true, bookmarks_parent: @owner))
-            @filtering_facets = @search.search_results.facets
           end
           @bookmarks = @search.search_results
           @facets = @bookmarks.facets
+          if @search.options[:excluded_tag_ids].present?
+            tags = Tag.where(id: @search.options[:excluded_tag_ids])
+            excluded_bookmark_tag_ids = params.dig(:exclude_bookmark_search, :tag_ids) || []
```

Comment:
Bit of an aside, but this line makes me think that the code above for packing all the information from :exclude_bookmark_search into :excluded_tag_ids is kind of overzealous (and I think it makes the include/exclude somewhat asymmetric). Still, I think that's probably outside the scope of this pull request. Maybe something to refactor later.